### PR TITLE
[Spark load][Fe 6/6] Fe process etl and loading state job

### PR DESCRIPTION
**Pull request by wyb**
1. Fe checks the status of etl job regularly 
1.1 If status is RUNNING, update etl job progress
1.2 If status is CANCELLED, cancel load job
1.3 If status is FINISHED, get the etl output file paths, update job state to LOADING and log job update info

2. Fe sends PushTask to Be and commits transaction after all push tasks execute successfully

#3433

### CODE REVIEW

File: fe/src/main/java/org/apache/doris/load/loadv2/SparkLoadJob.java [hunk 1]
```diff
@@ -0,0 +1,700 @@
...
+                destSlotDesc.setIsMaterialized(true);
+                destSlotDesc.setColumn(column);
+                if (column.isAllowNull()) {
+                    destSlotDesc.setIsNullable(true);
+                } else {
+                    destSlotDesc.setIsNullable(false);
+                }
+            }
+            // Push broker scan node
+            PushBrokerScanNode scanNode = new PushBrokerScanNode(destTupleDesc);
```
Comment (imay) [inline fe/src/main/java/org/apache/doris/load/loadv2/SparkLoadJob.java:156]: Why is this ScanNode needed?
Seems that this scanNode will be destruct after this function.

File: fe/src/main/java/org/apache/doris/load/loadv2/SparkLoadJob.java [hunk 2]
```diff
@@ -0,0 +1,700 @@
...
+    private Map<String, Pair<String, Long>> tabletMetaToFileInfo = Maps.newHashMap();
+
+    // --- members below not persist ---
+    // temporary use
+    // one SparkLoadJob has only one table to load
+    // hivedb.table for global dict
+    private String hiveTableName = "";
+    private ResourceDesc resourceDesc;
+    // for spark standalone
+    private SparkAppHandle sparkAppHandle;
```
Comment (imay) [inline fe/src/main/java/org/apache/doris/load/loadv2/SparkLoadJob.java:123]: When is this assigned?

File: fe/src/main/java/org/apache/doris/load/loadv2/SparkLoadJob.java [hunk 3]
```diff
@@ -0,0 +1,700 @@
...
+        // log loading state
+        logUpdateStateInfo();
+
+        // create push tasks
+        prepareLoadingInfos();
+        submitPushTasks();
+    }
+
+    private void updateToLoadingState(EtlStatus etlStatus, Map<String, Long> filePathToSize) throws LoadException {
+        writeLock();
```
Comment (imay) [inline fe/src/main/java/org/apache/doris/load/loadv2/SparkLoadJob.java:354]: why not get lock in the outer public function? In most scenario, there is only one writer.
If you put this lock in a private function, there is a case that multiple threads will do proceeEtlFinish concurrently.

File: fe/src/main/java/org/apache/doris/load/loadv2/SparkLoadJob.java (hunk 2 above)
Comment (wyb) [reply fe/src/main/java/org/apache/doris/load/loadv2/SparkLoadJob.java:126]: setJobProperties in SparkLoadJob.
This is for temporary use. I am investigating load from hive table, and i will update it recently.

**Review by morningman — DISMISSED** [review]
LGTM

**Review by morningman — APPROVED** [review]
LGTM

**Comment by morningman** [comment]
The compile error of broker has been fix in #3913 .
So I just approve this PR.

**Pull request merged**